stock_putaway_storage_type_strategy/models/location_storage_type.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class StockLocationStorageType(models.Model):
+
+    _inherit = 'stock.location.storage.type'
+
+    only_empty = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if there are not any existing quant nor planned move on "
+             "this location")
+    do_not_mix_lots = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if the location contains product of the same "
+             "stock.production.lot"
+    )
+    do_not_mix_products = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if the location contains the same product."
+    )
+    # max_height = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging height is lower than this maximum."
+    # )
+    # max_height_value = fields.Float()
+    # max_volume = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging volume is lower than this maximum."
+    # )
+    # max_volume_value = fields.Float()
+
+    @api.constrains('only_empty', 'do_not_mix_lots', 'do_not_mix_products')
+    def _check_empty_mix(self):
+        for location_storage_type in self:
+            if (
```

Comment:
I like
```suggestion
            if any(location_storage_type.only_empty and (location_storage_type.do_not_mix_lots or 
                location_storage_type.do_not_mix_product) for location_storage_type in self):
```

 to avoid an indentation, but that's me.